cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterContractCatalogController.java

```diff
@@ -0,0 +1,127 @@
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.watersupply.WaterUserContract;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public class WaterContractCatalogController implements Handler {
+    private final Logger LOGGER = Logger.getLogger(WaterContractCatalogController.class.getName());
+    public static final String TAG = "Water Contracts";
+    private static final String WATER_USER = "water-user";
+    private final MetricRegistry metrics;
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    public WaterContractCatalogController(MetricRegistry metrics) {
+        this.metrics = metrics;
+    }
+
+    @NotNull
+    protected WaterContractDao getContractDao(DSLContext dsl) {
+        return new WaterContractDao(dsl);
+    }
+
+    @OpenApi(
+        pathParams = {
+            @OpenApiParam(name = OFFICE, description = "Specifies the"
+                    + " office that the contract is associated with.", required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "Specifies the project id of the contract.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "Specifies the water user of the contract.",
+                    required = true),
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_200,
+                content = {
+                    @OpenApiContent(from = WaterUserContract.class, type = Formats.JSONV1),
+                    @OpenApiContent(from = WaterUserContract.class, type = Formats.JSON)
+                }),
+            @OpenApiResponse(status = "404", description = "The provided combination of parameters"
+                    + " did not find any contracts."),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Return all water contracts",
+        path = "/projects/{office}/{project-id}/water-users/{water-user}/contracts",
+        method = HttpMethod.GET,
+        tags = {TAG}
+    )
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            final String office = ctx.queryParam(OFFICE);
```

Comment:
Office and name are marked as path parameters in the swagger docs, but you have them as query parameters here.

Reply:
Switched to path parameters